src/main/java/com/icthh/xm/tmf/ms/communication/channel/twilio/TwilioChannelHandler.java

```diff
@@ -0,0 +1,135 @@
+package com.icthh.xm.tmf.ms.communication.channel.twilio;
+
+import com.icthh.xm.commons.topic.config.MessageListenerContainerBuilder;
+import com.icthh.xm.commons.topic.domain.ConsumerHolder;
+import com.icthh.xm.commons.topic.domain.TopicConfig;
+import com.icthh.xm.tmf.ms.communication.channel.ChannelHandler;
+import com.icthh.xm.tmf.ms.communication.channel.telegram.KafkaToTelegramMessageHandler;
+import com.icthh.xm.tmf.ms.communication.config.ApplicationProperties;
+import com.icthh.xm.tmf.ms.communication.domain.CommunicationSpec;
+import com.icthh.xm.tmf.ms.communication.domain.CommunicationSpec.Channels;
+import com.icthh.xm.tmf.ms.communication.domain.CommunicationSpec.Telegram;
+import com.icthh.xm.tmf.ms.communication.domain.MessageType;
+import com.icthh.xm.tmf.ms.communication.service.TelegramService;
+import com.icthh.xm.tmf.ms.communication.service.TwilioService;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.time.StopWatch;
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.listener.AbstractMessageListenerContainer;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class TwilioChannelHandler implements ChannelHandler {
```

Comment:
is it possible to extract common properties and logic from TwilioChannelHandler and TelegramChannelHandler to Abstract class?

Reply:
Will be refactored after @sergeysenja1992 MR